### ISSUE TITLE
Too many files open error reported when file system watching enabled

### ISSUE THREAD

**Issue by mark-vieira** [gradle/gradle#13447]
### Current Behavior
When using the new experimental file system watching support via `org.gradle.unsafe.watch-fs=true` occassionally get build errors reporting too many files open.

### Context

For example:

```
Exception: java.io.FileNotFoundException: /Users/mark/.gradle/build-scan-data/3.3.3/pending-uploads/1592330774545-ab207c50-b059-4921-b789-df35ebb82c90-upload.scan (Too many open files)
```

and then in a subsequent build with the same daemon:

```
* What went wrong:
Execution failed for task ':buildSrc:compileMinimumRuntimeJava'.
> Could not resolve all dependencies for configuration ':buildSrc:minimumRuntimeCompileClasspath'.
   > java.io.FileNotFoundException: /private/var/folders/s7/gryzdw8d5hv8m4kwldldp5nr0000gn/T/gradle16770923741351733224.bin (Too many open files)
```

Using `gradle --stop` to kill the daemon seems to resolve the issue, at least temporarily.
### Your Environment

This is happening on MacOS 10.15.5 using OpenJDK 14.

**Comment by lptr** [comment]
Can you please giver me your file limits? `launchctl limit`

**Comment by lptr** [comment]
The expectation is that this PR will help remedy any "too many open files" problems on macOS: https://github.com/gradle/gradle/pull/13222

**Comment by mark-vieira** [comment]
```
$ launchctl limit
	cpu         unlimited      unlimited      
	filesize    unlimited      unlimited      
	data        unlimited      unlimited      
	stack       8388608        67104768       
	core        0              unlimited      
	rss         unlimited      unlimited      
	memlock     unlimited      unlimited      
	maxproc     5568           8352           
	maxfiles    256            unlimited
```

**Comment by lptr** [comment]
@mark-vieira our understanding is that this problem is fixed in 6.6 since #13222 is fixed. I'm going to close this for now. Please reopen if the problem persists.
